### PR TITLE
chore: add test to check if big int datatype messes with the proxy

### PR DESCRIPTION
**Pull request by KDederichs**
related to #710

### CODE REVIEW

**Comment by nikophil** [comment]
Aaaand tests are passing 😅

(The failure comes from a namespace problem)

**Comment by KDederichs** [comment]
yeah I'm trying to figure out if it's a Postgres thing... but something's wrong with those, can't figure out what atm :D

**Comment by KDederichs** [comment]
There you go, it only happens when you use an UUID as primary key... figures (no clue how to fix the other tests, not sure what's wrong with them)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -31,6 +31,7 @@ jobs:
           - {php: 8.3, symfony: '*', phpunit: 9, database: none}
           - {php: 8.3, symfony: '*', phpunit: 11, database: mysql|mongo}
           - {php: 8.3, symfony: '*', phpunit: 11, database: pgsql|mongo}
+          - {php: 8.3, symfony: '*', phpunit: 9, database: pgsql}
```
Comment (nikophil) [inline .github/workflows/ci.yml:34]: seems unneeded

File: tests/Fixture/Factories/BigIntProxyFactory.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+use Zenstruck\Foundry\Tests\Fixture\Entity\BigIntEntity;
+use Zenstruck\Foundry\Persistence\PersistentProxyObjectFactory;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ *
+ * @extends PersistentProxyObjectFactory<BigIntEntity>
+ */
+final class BigIntProxyFactory extends PersistentProxyObjectFactory
```
Comment (nikophil) [inline tests/Fixture/Factories/BigIntProxyFactory.php]: FYI, you can use [anonymous factories](https://symfony.com/bundles/ZenstruckFoundryBundle/current/index.html#anonymous-factories)

**Review by nikophil — COMMENTED** [review]
this is very strange, that it only fails when uuid is involved :shrug:

**Comment by nikophil** [comment]
ok then, it has nothing do to with uuid. If you do this change:
```diff
- self::assertEquals($entity->getId(), $entity->getId());
+ self::assertEquals($entity->getBigIntVal(), $entity->getBigIntVal());
```
it will fail regardeless you use uuid or not.

And it receives an int from the database, so it thinks the original data is an int... this looks like a little quirk in the ORM.

I'm afraid there is nothing we can do in Foundry. I'll close this PR, and keep the issue for the record.

**Comment by KDederichs** [comment]
Yeah I suspect it might be a doctrine issue, cause it's supposed to load that as string according to them, I guess I'll bring it up there and see what they say about it.

**Pull request closed without merging**